src/modules/visitor-space/components/MaterialRequestBlade/MaterialRequestBlade.tsx

```diff
@@ -0,0 +1,240 @@
+import { Button, RadioButton, TextArea } from '@meemoo/react-components';
+import clsx from 'clsx';
+import Image from 'next/image';
+import React, { FC, useState } from 'react';
+
+import { MaterialRequestsService } from '@material-requests/services';
+import { MaterialRequestType } from '@material-requests/types';
+import { Blade, Icon, IconNamesLight } from '@shared/components';
+import useTranslation from '@shared/hooks/use-translation/use-translation';
+import { toastService } from '@shared/services/toast-service';
+
+import styles from './MaterialRequestBlade.module.scss';
+
+interface MaterialRequestBladeProps {
+	isOpen: boolean;
+	onClose: () => void;
+	objectName: string;
+	objectId: string;
+	objectType: string;
+	maintainerName: string;
+	maintainerLogo: string;
+	maintainerSlug: string;
+}
+
+const MaterialRequestBlade: FC<MaterialRequestBladeProps> = ({
+	isOpen,
+	onClose,
+	objectName,
+	objectId,
+	objectType,
+	maintainerName,
+	maintainerLogo,
+	maintainerSlug,
+}) => {
+	const { tText } = useTranslation();
+
+	const [typeSelected, setTypeSelected] = useState<MaterialRequestType>(MaterialRequestType.VIEW);
+	const [reasonInputValue, setReasonInputValue] = useState('');
+	const [showError, setShowError] = useState(false);
+
+	const onCloseModal = () => {
+		onClose();
+		setReasonInputValue('');
+		setTypeSelected(MaterialRequestType.VIEW);
+		setShowError(false);
+	};
+
+	const onAddToList = async () => {
+		try {
+			if (reasonInputValue.length > 0) {
+				const response = await MaterialRequestsService.create({
+					objectId,
+					type: typeSelected,
+					reason: reasonInputValue,
+					requesterCapacity: 'OTHER',
+				});
+				if (response === undefined) {
+					return;
```

Comment:
if response is empty something went wrong, show an error toast message, so the user knows something went wrong.
Best to leave the blade open when something went wrong in case they wrote an extensive "reason" in the reason field, so they can copy it, and try again.